### PR TITLE
Separate NSJ Converter in a Companion Package

### PR DESCRIPTION
**Pull request by vmelamed**
# Summary

Separate NSJ converter in a companion Package to preserve AoT compatibility in the main package. 
BREAKING CHANGE!

## Commits

- fix: enable AoT, refactor Directory.Build.props
- chore: update vm2.TestUtilities package version to 1.5.2 and clarify package reference comments
- refactor!: put the Newtonsoft.Json Ulid converter in a separate project/assembly, allowing for AoT in the main project and non-AoT in that converter only -- BREAKING CHANGE. Updated with diff-shared.sh.

## Checklist

- [x] Commit messages follow [Conventional Commits](https://www.conventionalcommits.org/) format
- [x] Tests pass locally, and cover new code and edge cases. Test coverage is above 80%.
- [x] It is not expected that the performance will degrade by more than 20%
- [x] Documentation is updated if necessary

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -1,4 +1,35 @@
...
+            ">true</IsTestLibraryProject>
+
+        <!-- Property to be used in conditions for detecting benchmark tests -->
+        <IsInBenchmarksFolder Condition="
+            $([System.Text.RegularExpressions.Regex]::IsMatch($(MSBuildProjectDirectory), '[/\\]benchmarks?[/\\]', RegexOptions.IgnoreCase))
+            ">true</IsInBenchmarksFolder>
+
+        <!-- Property to be used in conditions for detecting example usage projects -->
+        <IsInExamplesFolder Condition="
+            $([System.Text.RegularExpressions.Regex]::IsMatch($(MSBuildProjectDirectory), '[/\\]examples?[/\\]', RegexOptions.IgnoreCase))
```
[the remark this answers is not in the document]
Comment (vmelamed) [reply Directory.Build.props]: Works fine everywhere else.

File: README.md [hunk 2]
```diff
@@ -276,12 +276,18 @@ e.g. for testing purposes, for performance reasons, or if you are concerned abou
 can explicitly specify that the factory should use the pseudo-random number generator `vm2.UlidRandomProviders.PseudoRandom`.
 You can also provide your own, thread-safe implementation of `vm2.IRandomNumberGenerator` to the factory.
 
-#### Timestamp Provider (`vm2.ITimestampProvider`)
+#### Timestamp Provider (`System.TimeProvider`)
 
-By default, the timestamp provider uses `DateTimeOffset.UtcNow` converted to Unix epoch time in milliseconds. If you need a
-different source of time, e.g. for testing purposes, you can provide your own implementation of `vm2.ITimestampProvider` to the
+By default, the timestamp provider uses `System.TimeProvider.System.GetUtcNow().ToUnixTimeMilliseconds()` converted to Unix epoch time in milliseconds. If you need a different source of time, e.g. for testing purposes, you can override the .NET BCL class `System.TimeProvider` or for testing purposes `System.FakeTimeProvider` to provide your own implementation.
 factory.
```
[the remark this answers is not in the document]
Comment (vmelamed) [reply README.md]: fixed

File: test/UlidType.Tests/NsJson/UlidNsConverterTests.cs [hunk 3]
```diff
@@ -3,13 +3,16 @@
...
-public class UlidNsConverterTests
+public class UlidNsConverterTests : TestBase
 {
+    public UlidNsConverterTests(ITestOutputHelper output) : base(output)
+    {
+        var settings = new JsonSerializerSettings();
+        settings.Converters.Add(new UlidNsConverter());
+        JsonConvert.DefaultSettings = () => settings;
+    }
+
```
[the remark this answers is not in the document]
Comment (vmelamed) [reply test/UlidType.Tests/NsJson/UlidNsConverterTests.cs]: refactored
